File: front-for-restaurant/src/app/summary/invoice.ts
export class Invoice {
  private id: number;
  private date: Date;
  private moneyThatNeedToBePaid: number;
  moneyReceived: number;
  private rest: number;

  constructor(id: number, date: Date, moneyThatNeedToBePaid: number, moneyReceived: number, rest: number) {
    this.id = id;
    this.date = date;
    this.moneyThatNeedToBePaid = moneyThatNeedToBePaid;
    this.moneyReceived = moneyReceived;
    this.rest = rest;
  }
  getMoney(): number {
    return this.moneyReceived;
  }
}

File: backend/src/main/java/com/example/restaurant/domain/invoice/Invoice.java
package com.example.restaurant.domain.invoice;

import lombok.Value;

import java.util.Date;

@Value
public class Invoice {
    private final String id;
    private final Date date;
    private final Double moneyThatNeedToBePaid;
    private final Double moneyReceived;

    public Double getRest(){
        return moneyReceived -moneyThatNeedToBePaid;
    }
}
